tests/wast.rs

```diff
@@ -10,15 +11,188 @@ use wasmtime::{
+            continue;
+        }
+
+        for strategy in [Strategy::Cranelift, Strategy::Winch] {
+            for pooling in [true, false] {
+                let trial = Trial::test(
+                    format!(
+                        "{strategy:?}/{}{}",
+                        if pooling { "pooling/" } else { "" },
+                        path.to_str().unwrap()
+                    ),
+                    {
+                        let path = path.clone();
+                        move || {
+                            run_wast(&path, strategy, pooling).map_err(|e| format!("{e:?}").into())
+                        }
+                    },
+                );
+                trials.push(trial.with_ignored_flag(ignore(&path, strategy)));
+            }
+        }
+    }
+}
+
+fn ignore(test: &Path, strategy: Strategy) -> bool {
+    if cfg!(miri) {
+        return true;
+    }
+
+    // Winch only supports x86_64 at this time.
+    if strategy == Strategy::Winch && !cfg!(target_arch = "x86_64") {
+        return true;
+    }
+
+    for part in test.iter() {
+        // Not implemented in Wasmtime yet
+        if part == "exception-handling" {
+            return true;
+        }
+        // Not implemented in Wasmtime yet
+        if part == "extended-const" {
+            return true;
+        }
+
+        // TODO(#6530): These tests require tail calls, but s390x doesn't
+        // support them yet.
+        if cfg!(target_arch = "s390x") {
+            if part == "function-references" || part == "tail-call" {
+                return true;
+            }
+        }
+
+        // Disable spec tests for proposals that Winch does not implement yet.
+        if strategy == Strategy::Winch {
+            let part = part.to_str().unwrap();
+            if false
+                // wasm proposals that Winch doesn't support
+                || part == "references"
+                || part == "tail-call"
+                || part == "gc"
+                || part == "threads"
+                || part == "multi-memory"
+                // || part == "simd"
+                || part == "relaxed-simd"
+                || part == "function-references"
+                // tests in misc_testsuite that Winch doesn't support
+                || part == "no-panic.wast"
+                || part == "externref-id-function.wast"
+                || part == "int-to-float-splat.wast"
+                || part == "issue6562.wast"
+                || part == "many_table_gets_lead_to_gc.wast"
+                || part == "mutable_externref_globals.wast"
+                || part == "no-mixup-stack-maps.wast"
+                || part == "simple_ref_is_null.wast"
+                || part == "table_grow_with_funcref.wast"
+                // Tests in the spec test suite Winch doesn't support
+                || part.starts_with("simd")
+                || part.starts_with("ref_")
+                || part == "threads.wast"
+				|| part == "br_table.wast"
```

Comment:
What happened to the indentation here? Anyway, can I suggest either using `["references", "tail-call", ...].contains(part)` or `match part { "references" | "tail-call" | ... }`, with the `starts_with` cases separate?

Reply:
Looks like some hard tabs snuck in by accident and rustfmt gives up in comments-interleaved-in-`if` condition. In any case your suggestion is much better!